validator/src/journal/block_manager.rs

```diff
@@ -557,20 +557,59 @@ impl Iterator for GetBlockIterator {
 
 pub struct BranchIterator {
     state: Arc<RwLock<BlockManagerState>>,
+    initial_block_id: String,
     next_block_id: String,
     blockstore: Option<String>,
 }
 
 impl BranchIterator {
     fn new(state: Arc<RwLock<BlockManagerState>>, first_block_id: String) -> Self {
+        let next_block_id = {
+            let mut block_manager = state
+                .write()
+                .expect("No log holder should have poisoned the lock");
+            match block_manager.ref_block(&first_block_id) {
+                Ok(_) => first_block_id.clone(),
+                Err(BlockManagerError::UnknownBlock) => NULL_BLOCK_IDENTIFIER.to_string(),
+
+                Err(err) => {
+                    error!(
+                        "Unable to ref block at {}: {:?}; ignoring",
+                        &first_block_id, err
+                    );
+                    NULL_BLOCK_IDENTIFIER.to_string()
```

Comment:
Handling this without panic'ing makes sense in the context of the existing API for the branch_diff iterator, but it raises the question whether the return type here should be `Result<Self, Error>` so we can return an error when the tip isn't found, rather than returning an empty iterator.